fendermint/vm/interpreter/src/fvm/exec.rs

```diff
@@ -48,7 +50,7 @@ where
     /// Return validator power updates and the next base fee.
     /// Currently ignoring events as there aren't any emitted by the smart contract,
     /// but keep in mind that if there were, those would have to be propagated.
-    type EndOutput = (PowerUpdates, BlockGasLimit);
+    type EndOutput = (PowerUpdates, BlockGasLimit, CurrentValidators);
```

Comment:
I think this approach is not correct, because essentially you are query the current list of validators at every block end. Then what is the point of maintaining a cache? Also most of the time, there is no validator change, so the querying of validator list is not necessary.

I would say returning the list of current validators when PowerUpdates is non empty would be more effecient.